app/store/middleware.js

```diff
@@ -425,3 +497,5 @@ function removePendingPost(pendingPostIds, id) {
         pendingPostIds.splice(pendingIndex, 1);
     }
 }
+
+export const middlewares = [messageRetention, saveState];
```

Comment:
Maybe if you  do the following Android won't even try and execute it

```suggestion
const middlewares = [messageRetention];
if (Platform.OS === 'ios') {
    middlewares.push(saveState);
}
export middlewares;
```

And then you  don't need to check for the platform inside the function

Reply:
> Another suggestion would be changing the name of `saveState` middleware to something that is more clear on what it does, but I couldn't think of anything :P

Done. `saveState` -> `saveShareExtensionState`.

Reply:
> Maybe if you do the following Android won't even try and execute it
> 
> And then you don't need to check for the platform inside the function

The only trouble with this is that it messes with the sequence/order that the middlewares get executed, and I expect we'll be adding at least one more middleware piece in an upcoming PR. There are no dependencies between middlewares, but if run in a different order, some middlewares may end up doing more work than necessary.

Reply:
As far as I'm seeing the order would be preserved, right
1. messageRetention
2. saveShareExtensionState

Reply:
Correct. Trouble comes when we add a third and fourth middleware. How would we ensure then that iOS-specific middlewares go into the appropriate spot in the order (in this case 2nd)?

Reply:
in that case you can use `splice` as in `middlewares.splice(1, 0, saveShareExtenstionState)`

Reply:
Done. I went with the original suggestion (using `push`) because we really should not have any order-based dependencies at all, and this iOS-specific middleware in question is likely to disappear soon with managed state persistence around the corner.